### PR TITLE
[ADPRODUCTS-367] Updates to tiles clients daily

### PR DESCRIPTION
**Pull request by skahmann3**
Checklist for reviewer:

Fix to repeated rows for mobile clients. Fix to NULL values for desktop clicks + impressions.

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

**Review by rebecca-burwei — APPROVED** [review]
Looks good to me :)

File: sql/moz-fx-data-shared-prod/telemetry_derived/sponsored_tiles_clients_daily_v1/query.sql [hunk 1]
```diff
@@ -112,22 +112,38 @@ ios_events AS (
...
-    `mozfun.norm.browser_version_info`(client_info.app_display_version) AS browser_version_info,
-    mozfun.glean.legacy_compatible_experiments(ping_info.experiments) AS experiments,
-    normalized_country_code AS country,
-    client_info.locale,
-    normalized_channel,
-    normalized_os_version,
-    sample_id
+    ARRAY_AGG(DATE(submission_timestamp) ORDER BY submission_timestamp DESC)[
+      OFFSET(0)
+    ] AS submission_date,
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/telemetry_derived/sponsored_tiles_clients_daily_v1/query.sql:117]: You could replace all these `ARRAY_AGG` statements with

```
FIRST_VALUE(x) OVER (PARTITION BY client_id ORDER BY submission_timestamp DESC)
```

You'd also have to throw a `SELECT DISTINCT` at the top, which is too bad.

Maybe it's just personal preference, but that feels to me more like the preferred way to do it.

File: sql/moz-fx-data-shared-prod/telemetry_derived/sponsored_tiles_clients_daily_v1/query.sql [hunk 2]
```diff
@@ -112,22 +112,38 @@ ios_events AS (
...
+    ARRAY_AGG(client_info.app_display_version ORDER BY submission_timestamp DESC)[
+      OFFSET(0)
+    ] AS browser_version_info,
+    mozfun.map.mode_last(
+      ARRAY_CONCAT_AGG(
+        mozfun.glean.legacy_compatible_experiments(ping_info.experiments)
+        ORDER BY
+          submission_timestamp
+      )
+    ) AS experiments,
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/telemetry_derived/sponsored_tiles_clients_daily_v1/query.sql:128]: Oh except here, I guess, this seems quite a bit more complicated. What's happening here? I don't totally follow

File: sql/moz-fx-data-shared-prod/telemetry_derived/sponsored_tiles_clients_daily_v1/query.sql [hunk 3]
```diff
@@ -159,33 +175,54 @@ android_events AS (
...
   SELECT
-    date(submission_timestamp) AS submission_date,
-    client_info.client_id,
-    `mozfun.norm.browser_version_info`(client_info.app_display_version) AS browser_version_info,
-    mozfun.glean.legacy_compatible_experiments(ping_info.experiments) AS experiments,
-    normalized_country_code AS country,
-    client_info.locale,
-    normalized_channel,
-    normalized_os_version,
-    -- profile_age_in_days,
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/telemetry_derived/sponsored_tiles_clients_daily_v1/query.sql:180]: We're good to get rid of this commented out line, I guess? It's not here for a special reminder or something?

**Review by curtismorales — COMMENTED** [review]
This looks like it does what you want it to do, I just want to chat with you about the ARRAY_AGG quickly

**Review by curtismorales — APPROVED** [review]

**Pull request merged**